.github/mergify.yml

```
@@ -58,26 +143,313 @@ pull_request_rules:
 
   - name: "ask to resolve conflict"
     conditions:
+      - and: *is_open
       - "conflict"
     actions:
       comment:
         message: "This pull request now has conflicts. Could you fix it @{{author}}? 🙏"
+      label:
+        toggle:
+          - conflict
+
+  - name: "ask to not edit the readme"
+    conditions:
+      - and: *is_open
+      - files=README.md
+      - -files=README.yaml
+    actions:
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > Do not edit the `README.md` directly. It's auto-generated from the `README.yaml`
+          >
+
+          Rebuild the `README.md` by running `make readme` and commit the changes.
+
+          ```shell
+          make init
+          make readme
+          ```
+
+          Could you fix it @{{author}}? 🙏
+
+  - name: "ask to rebuild readme"
+    conditions:
+      - and: *is_open
+      - files=README.yaml
+      - -files=README.md
+    actions:
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > `README.md` is out of date.
+          >
+
+          Rebuild the `README.md` by running `make readme` and commit the changes.
+
+          ```shell
+          make init
+          make readme
+          ```
+
+          Could you fix it @{{author}}? 🙏
+
+  - name: "ask for title"
+    conditions:
+      - and: *is_open
+      - -title~=^[0-9A-Za-z]+
+    actions:
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > ## Title is necessary and should not be empty.
+          >
+          > Kindly provide a meaningful title for this Pull Request.
+
+  - name: "ask for description"
+    conditions:
+      - and: *is_open
+      - -body~=[0-9A-Za-z]{3,}\\s+[0-9A-Za-z]{3,}\\s+[0-9A-Za-z]{3,}
+      - body~=(Describe high-level what changed)
+    actions:
+      comment:
+        message: |
+          > [!IMPORTANT]
+          > ## Description is necessary and should not be empty.
+          >
+          > Kindly provide details with **what** was changed, **why** it was changed.
 
   - name: "remove outdated reviews"
     conditions:
+      - and: *is_open
       - or: *default_branch
     actions:
       dismiss_reviews:
```

Would we need this, otherwise things like changing labels will dismiss reviews?

```suggestion
      dismiss_reviews:
        when: synchronize
```